usr/sbin/p11sak/p11sak.c

```diff
@@ -1117,6 +1150,151 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(name, CT_BAREVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", name->line);
+                    return CKR_DATA_INVALID;
+                } else if (hex_string == NULL) {
+                    fprintf(stderr, "Missing value in Attribute in line %hu\n", hex_string->line);
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(hex_string, CT_INTVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", hex_string->line);
+                    return CKR_DATA_INVALID;
+                } else if (type == NULL) {
+                    fprintf(stderr, "Missing type in Attribute in line %hu\n", type->line);
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(type, CT_BAREVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", type->line);
+                    return CKR_DATA_INVALID;
+                }
+                
+                // print attribute by type
+                unsigned int hex = confignode_to_intval(hex_string)->value;
+                if (strcmp(confignode_to_bareval(type)->value, "CK_BBOOL") == 0) {
+                    // build template
+                    CK_BBOOL a_bool;
+                    CK_ULONG bs = sizeof(CK_BBOOL);
+                    CK_ATTRIBUTE temp = {hex, &a_bool, bs};
+
+                    // get attribute value
+                    rc = funcs->C_GetAttributeValue(session, hkey, &temp, 1);
+                    if (rc == CKR_OK) {
+                        if (temp.ulValueLen != sizeof(CK_BBOOL)) {
+                            printf(" Error in retrieving Attribute %s: %lu\n",
+                                    confignode_to_bareval(name)->value, temp.ulValueLen);
+                        } else {
+                            if (long_print) {
+                                printf("          %s: %s\n", confignode_to_bareval(name)->value,
+                                        CK_BBOOL2a(*(CK_BBOOL*) temp.pValue));
+                            } else if (strcmp(CK_BBOOL2a(*(CK_BBOOL*) temp.pValue), "CK_TRUE") == 0) {
+                                printf(" 1 ");
+                                return rc;
+                            }
+                        }    
+                    } else if (rc == CKR_ATTRIBUTE_SENSITIVE) {
+                        printf("          %s: SENSITIVE\n", confignode_to_bareval(name)->value);
+                    } else if (rc != CKR_ATTRIBUTE_TYPE_INVALID) {
+                        return rc;
+                    } 
+                } else if (strcmp((confignode_to_bareval(type)->value), "CK_ULONG") == 0) {
+                    // build template
+                    CK_ULONG a_ulong;
+                    CK_ULONG us = sizeof(CK_ULONG);
+                    CK_ATTRIBUTE temp = {hex, &a_ulong, us};
```

Comment:
variable `us` is superfluous, if you simply use
`CK_ATTRIBUTE temp = {hex, &a_ulong, sizeof(a_ulong)};`

Reply:
Done